ods_ci/tasks/Resources/RHODS_OLM/pre-tasks/oc_is_operator_installed.robot

```diff
@@ -49,17 +46,40 @@ Is RHODS Installed
   RETURN  ${result}
 
 Check A RHODS Family Operator Is Installed
-  [Documentation]   Returns if an operator with given subscription name has a CSV in the given namespace
-  [Arguments]    ${namespace}  ${subscription}
-  Log   Getting CSV from subscription ${subscription} namespace ${namespace}      console=yes
-  ${rc}    ${current_csv_name} =    Run And Return Rc And Output
-  ...    oc get subscription ${subscription} -n ${namespace} -ojson | jq '.status.currentCSV' | tr -d '"'
-  Log   Got CSV ${current_csv_name} from subscription ${subscription}, result: ${rc}      console=yes
-  IF  "${rc}" == "0" and "${current_csv_name}" != "${EMPTY}"
-      ${result} =  Run Keyword And Return Status
-      ...  Oc Get  kind=ClusterServiceVersion  namespace=${namespace}  name=${current_csv_name}
-  ELSE
-      ${result} =  Set Variable    False
+  [Documentation]   Returns if RHODS operator has a CSV
+
+  ${result} =  Set Variable    False
+  Log   Check if the RHODS operator installed.    console=yes
+  ${subscription}=    Get RHODS Subscription Name
+  ${namespace}=    Get RHODS Namespace
+  IF  "${subscription}" != "${EMPTY}"
+      Log   Getting CSV from subscription ${subscription} namespace ${namespace}      console=yes
+      ${rc}    ${current_csv_name} =    Run And Return Rc And Output
+      ...    oc get subscription ${subscription} -n ${namespace} -ojson | jq '.status.currentCSV' | tr -d '"'
+      Log   Got CSV '${current_csv_name}' from subscription '${subscription}', result: ${rc}      console=yes
+      IF  "${rc}" == "0" and "${current_csv_name}" != "${EMPTY}"
+          ${rc}=    Run Keyword And Return Status    oc get csv ${current_csv_name} -n ${namespace}
+          ${result}=    Evaluate    ${rc} == 0
+          Log   The csv exists: ${result}      console=yes
+      END
   END
-  Log   Operator with sub ${subscription} is installed result: ${result}      console=yes
+  Log   Operator with sub '${subscription}' is installed result: ${result}      console=yes
   IF  not ${result}     FAIL    The operator with sub ${subscription} is not installed.
+
+Get RHODS Subscription Name
+    [Documentation]    Returns the subscription name of RHOAI/ODH operator
+    Log   Get the RHODS subscription name by package name: '${RHODS_PACKAGE_NAME}'    console=yes
+    ${rc}    ${out}=    Run And Return RC And Output
+    ...    oc get subscriptions -A | grep ${RHODS_PACKAGE_NAME} | awk '{print $2}'
+    Should Be Equal As Integers    ${rc}    0
+    Log   The RHODS Subscription Name is: '${out}'    console=yes
+    RETURN    ${out}
+
+Get RHODS Namespace
+    [Documentation]    Returns the namespace of RHOAI/ODH operator
+    Log   Get the RHODS namespace by package name: '${RHODS_PACKAGE_NAME}'    console=yes
+    ${rc}    ${out}=    Run And Return RC And Output
+    ...    oc get subscriptions -A | grep ${RHODS_PACKAGE_NAME} | awk '{print $1}'
```

Comment:
Using `grep` sometimes leads to multi-line results, for example
```
$ printf "rhods-operator\nrhods-operator-dev\n" | grep rhods-operator
rhods-operator
rhods-operator-dev
```
That's why I recommend using something like this
```
$ oc get sub -A -o json | jq --arg pkgName "${RHODS_PACKAGE_NAME}" -r '.items[] | select(.spec.name==$pkgName) | .metadata.namespace'
```

Reply:
Updated!!